js/client.js

```diff
@@ -1764,56 +1806,25 @@
 			var passedCurSideRoom = false;
 
 			var notificationCount = 0;
-			for (var id in app.rooms) {
-				if (app.rooms[id].notifications) notificationCount++;
-				if (!id || id === 'teambuilder' || id === 'ladder') continue;
-				var room = app.rooms[id];
-				var name = '<i class="fa fa-comment-o"></i> <span>' + (Tools.escapeHTML(room.title) || (id === 'lobby' ? 'Lobby' : id)) + '</span>';
-				if (id.substr(0, 7) === 'battle-') {
-					name = Tools.escapeHTML(room.title);
-					var formatid = id.substr(7).split('-')[0];
-					if (!name) {
-						var p1 = (room && room.battle && room.battle.p1 && room.battle.p1.name) || '';
-						var p2 = (room && room.battle && room.battle.p2 && room.battle.p2.name) || '';
-						if (p1 && p2) {
-							name = '' + Tools.escapeHTML(p1) + ' v. ' + Tools.escapeHTML(p2);
-						} else if (p1 || p2) {
-							name = '' + Tools.escapeHTML(p1) + Tools.escapeHTML(p2);
-						} else {
-							name = '(empty room)';
-						}
-					}
-					name = '<i class="text">' + formatid + '</i><span>' + name + '</span>';
-				}
-				if (room.isSideRoom) {
-					if (id !== 'rooms') {
-						sideBuf += '<li><a class="button' + (curId === id || curSideId === id ? ' cur' : '') + room.notificationClass + ' closable" href="' + app.root + id + '">' + name + '</a><a class="closebutton" href="' + app.root + id + '"><i class="fa fa-times-circle"></i></a></li>';
-						if (curSideId) {
-							// get left/right for side rooms
-							if (curSideId === id) {
-								passedCurSideRoom = true;
-							} else if (!passedCurSideRoom) {
-								this.curSideRoomLeft = id;
-							} else if (!this.curSideRoomRight) {
-								this.curSideRoomRight = id;
-							}
-						} else {
-							// get left/right
-							if (curId === id) {
-								passedCurRoom = true;
-							} else if (!passedCurRoom) {
-								this.curRoomLeft = id;
-							} else if (!this.curRoomRight) {
-								this.curRoomRight = id;
-							}
-						}
+			if (app.roomList.length) buf += '<ul>';
+			for (var i = 0; i < app.roomList.length; i++) {
+				var room = app.roomList[i];
+				if (room.notifications) notificationCount++;
+				var id = room.id;
+				var formatid = id.split('-')[1];
+				var name = Tools.escapeHTML(room.title);
+				if (!name) {
```

Comment:
`renderRoomTab` is almost perfect except it can't deal with the Rooms room, which must be rendered if the app supports it even if it doesn't exist yet. (I thought about changing it to take an id but that would cause problems in the Teambuilder and Ladder case.)